tests/foreman/cli/test_computeresource_vmware.py

```diff
@@ -11,158 +11,181 @@
         }
     )
-    assert vmware_cr['name'] == cr_name
-    assert vmware_cr['server'] == vmware.vmware_server
+    return vmware_cr
+
+
+@pytest.fixture
+def module_vmware_hostgroup(
+    module_vmware_cr,
+    module_provisioning_sat,
+    module_sca_manifest_org,
+    module_location,
+    default_architecture,
+    module_provisioning_rhel_content,
+    module_lce_library,
+    default_partitiontable,
+    module_provisioning_capsule,
+    pxe_loader,
+):
+    return module_provisioning_sat.sat.api.HostGroup(
+        name=gen_string('alpha'),
+        organization=[module_sca_manifest_org],
+        location=[module_location],
+        architecture=default_architecture,
+        domain=module_provisioning_sat.domain,
+        content_source=module_provisioning_capsule.id,
+        content_view=module_provisioning_rhel_content.cv,
+        compute_resource=module_vmware_cr,
+        kickstart_repository=module_provisioning_rhel_content.ksrepo,
+        lifecycle_environment=module_lce_library,
+        root_pass=settings.provisioning.host_root_password,
+        operatingsystem=module_provisioning_rhel_content.os,
+        ptable=default_partitiontable,
+        subnet=module_provisioning_sat.subnet,
+        pxe_loader=pxe_loader.pxe_loader,
+        group_parameters_attributes=[
+            # assign AK in order the hosts to be subscribed
+            {
+                'name': 'kt_activation_keys',
+                'parameter_type': 'string',
+                'value': module_provisioning_rhel_content.ak.name,
+            },
+        ],
+    ).create()
 
 
 @pytest.mark.tier1
-def test_positive_create_with_org(vmware):
-    """Create VMware Compute Resource with organizations
+@pytest.mark.upgrade
+def test_positive_create_with_org_and_loc(target_sat, module_org, module_location):
+    """Create VMware Compute Resource with organizations and locations
 
-    :id: 807a1f70-4cc3-4925-b145-0c3b26c57559
+    :id: 96faae3f-bc64-4147-a9fc-09c858e0a68f
 
     :customerscenario: true
 
     :expectedresults: VMware Compute resource is created
 
     :BZ: 1387917
 
-    :CaseAutomation: Automated
-
     :CaseImportance: Critical
     """
     cr_name = gen_string('alpha')
-    vmware_cr = make_compute_resource(
+    vmware_cr = target_sat.cli.ComputeResource.create(
         {
             'name': cr_name,
-            'organization-ids': vmware.org.id,
+            'organization-ids': module_org.id,
+            'location-ids': module_location.id,
             'provider': FOREMAN_PROVIDERS['vmware'],
-            'server': vmware.vmware_server,
-            'user': vmware.vmware_username,
-            'password': vmware.vmware_password,
-            'datacenter': vmware.vmware_datacenter,
+            'server': settings.vmware.vcenter,
+            'user': settings.vmware.username,
+            'password': settings.vmware.password,
+            'datacenter': settings.vmware.datacenter,
         }
     )
     assert vmware_cr['name'] == cr_name
```

Comment:
Can we additional checks for org and loc assigned to CR?

Reply:
Updated